src/sonic_ax_impl/mibs/ietf/rfc3433.py

```diff
@@ -0,0 +1,402 @@
+    SCALE = EntitySensorDataScale.UNITS
+    PRECISION = 4
+
+
+class XcvrRxPowerSensor(SensorInterface):
+    """
+    Transceiver rx power sensor.
+    (TYPE, SCALE, PRECISION) set according to SFF-8472
+    Sensor measures in range (0 W, +6.5535 mW) with step 1E-4 mW.
+    """
+
+    TYPE = EntitySensorDataType.WATTS
+    SCALE = EntitySensorDataScale.MILLI
+    PRECISION = 4
+
+    @classmethod
+    def mib_values(cls, raw_value):
+        """
+        NOTE: Here we convert rx power value from dB to mW.
+        Actually, rx power value in transceiver EEPROM is in watts,
+        but the value in STATE DB is in dB, so converting back.
+        """
+
+        # dBm -> mW convertion
+        converter = lambda x: 10 ** (x/10)
+        return super().mib_values(raw_value, converter=converter)
+
+
+class XcvrTxBiasSensor(SensorInterface):
+    """
+    Transceiver tx bias sensor.
+    (TYPE, SCALE, PRECISION) set according to SFF-8472
+    Sensor measures in range (0 mA, 131 mA) with step 2E-3 mA
+    """
+
+    TYPE = EntitySensorDataType.AMPERES
+    SCALE = EntitySensorDataScale.MILLI
+    PRECISION = 3
+
+
+# mapping between DB key and Sensor object
+TRANSCEIVER_SENSOR_MAP = {
+    "temperature": XcvrTempSensor,
+    "voltage":     XcvrVoltageSensor,
+    "rx1power":    XcvrRxPowerSensor,
+    "rx2power":    XcvrRxPowerSensor,
+    "rx3power":    XcvrRxPowerSensor,
+    "rx4power":    XcvrRxPowerSensor,
+    "tx1bias":     XcvrTxBiasSensor,
+    "tx2bias":     XcvrTxBiasSensor,
+    "tx3bias":     XcvrTxBiasSensor,
+    "tx4bias":     XcvrTxBiasSensor,
+}
+
+
+def get_transceiver_sensor(sensor_key):
+    """
+    Gets transceiver sensor object
+    :param sensor_key: Sensor key from XcvrDomDB
+    :param ifindex: Interface index associated with transceiver
+    :return: Sensor object.
+    """
+
+    return TRANSCEIVER_SENSOR_MAP[sensor_key]
+
+
+class PhysicalSensorTableMIBUpdater(MIBUpdater):
+    """
+    Updater for sensors.
+    """
+
+    def __init__(self):
+        """
+        ctor
+        """
+
+        super().__init__()
+
+        self.statedb = SonicV2Connector()
+        self.statedb.connect(self.statedb.STATE_DB)
+
+        # list of available sub OIDs
+        self.sub_ids = []
+
+        # sensor MIB requiered values
+        self.ent_phy_sensor_type_map = {}
+        self.ent_phy_sensor_scale_map = {}
+        self.ent_phy_sensor_precision_map = {}
+        self.ent_phy_sensor_value_map = {}
+        self.ent_phy_sensor_oper_state_map = {}
+
+    def reinit_data(self):
+        """
+        Reinit data, clear cache
+        """
+
+        # clear cache
+        self.ent_phy_sensor_type_map = {}
+        self.ent_phy_sensor_scale_map = {}
+        self.ent_phy_sensor_precision_map = {}
+        self.ent_phy_sensor_value_map = {}
+        self.ent_phy_sensor_oper_state_map = {}
+
+        transceiver_dom_keys = mibs.transceiver_dom_table("*")
+        self.transceiver_dom = self.statedb.keys(self.statedb.STATE_DB,
```

Comment:
>transceiver_dom [](start = 13, length = 15)

just decode them?

Reply:
Yes, now changed it to decode right in place